### PR TITLE
Cache the National aggregate data on redis level

### PR DESCRIPTION
**Pull request by Rohit25negi**
Changing the cache of the national aggregate result of fact sheet from memoised to icds_quickcache so that it can be cached on redis and we don't need to calculate it again when the request goes to different web worker

### CODE REVIEW

File: custom/icds_reports/reports/fact_sheets.py [hunk 1]
```diff
@@ -12,6 +11,40 @@
...
+        'previous_month': parse(previous_month),
+    }
+    datasource_argument = {
+        'config': national_config,
+        'loc_level': loc_level,
+        'show_test': show_test,
+    }
+
+    if data_source_name != 'AggCCSRecordMonthlyDataSource':
+        datasource_argument['beta'] = beta
```
Comment (czue) [inline custom/icds_reports/reports/fact_sheets.py:38]: not sure this is actionable, but it seems a little weird that the beta argument is not passed through to this report. would it make sense to add it as an (unused) argument? or that feels silly?

File: custom/icds_reports/reports/fact_sheets.py [hunk 2]
```diff
@@ -528,38 +561,21 @@ def new_table_config(self):
...
-            )
+        fact_sheet_datasources = dict()
+        datasource_arguments = {
+            'config': config,
+            'loc_level': self.loc_level,
+            'show_test': self.show_test
         }
+        for datasource_name, datasource_class in FACT_SHEET_DATASOURCES_NAMES.items():
+            arguments = datasource_arguments.copy()
+            if datasource_name != 'AggCCSRecordMonthlyDataSource':
```
Comment (czue) [inline custom/icds_reports/reports/fact_sheets.py:572]: the fact that it's repeated here makes me like the idea of adding it even more

File: custom/icds_reports/reports/fact_sheets.py [hunk 3]
```diff
@@ -662,8 +678,13 @@ def get_data(self):
                         row['data'].append({'html': 0})
 
                     if 'average' in row:
-                        data = self.get_data_for_national_aggregatation(
-                            row['data_source']
+                        data = get_data_for_national_aggregatation(
+                            self.config['domain'],
+                            self.config['previous_month'].strftime('%Y-%m-%d'),
```
Comment (czue) [inline custom/icds_reports/reports/fact_sheets.py:683]: we have to convert to string and back because quickcache can't handle dates?

**Review by czue — APPROVED** [review]
no blockers

File: custom/icds_reports/reports/fact_sheets.py (hunk 3 above)
Comment (Rohit25negi) [reply custom/icds_reports/reports/fact_sheets.py:683]: yes thats right

File: custom/icds_reports/reports/fact_sheets.py (hunk 2 above)
Comment (Rohit25negi) [reply custom/icds_reports/reports/fact_sheets.py:572]: yeah I think that makes more sense, I can add the beta param to the this class as well

**Pull request merged**